core/src/test/scala/io/aiven/guardian/kafka/Generators.scala

```diff
@@ -118,10 +118,35 @@ object Generators {
     def topics: Set[String] = data.map(_.topic).toSet
   }
 
+  /** Generates a random `FiniteDuration` that is between the time periods within the `reducedConsumerRecords`.
+    */
   def randomPeriodSliceBetweenMinMax(reducedConsumerRecords: List[ReducedConsumerRecord]): Gen[FiniteDuration] = {
     val head = reducedConsumerRecords.head
     val last = reducedConsumerRecords.last
-    Gen.choose[Long](head.timestamp, last.timestamp - 1).map(millis => FiniteDuration(millis, MILLISECONDS))
+    Gen.choose[Long](head.timestamp, last.timestamp).map(millis => FiniteDuration(millis, MILLISECONDS))
```

Comment:
I changed `head.timestamp, last.timestamp - 1` to `head.timestamp, last.timestamp ` because after implementing and testing `periodSliceLargerThanWholePeriodGenerated` I realized that the boundary is at `(last.timestamp - head.timestamp) + 1` and not `(last.timestamp - head.timestamp)`  so the `-1` here is pointless.